File: tests/thwap-tests.py
#!/usr/bin/env python
import sys
import os
import time
import random
import unittest

# Setup our library paths depending on whether or not we are being run by
# hand or from the Makefile.
if os.getenv('PWD').find('test') == -1:
	sys.path.insert(0, "%s/python/" % os.getenv('PWD'))
else:
	sys.path.insert(0, '%s/../python/' % os.getenv('PWD'))

# imports to test
from THWAP.core import config
from THWAP.os.linux.gentoo import glsa

# libTHWAP unit tests
class unitTest(unittest.TestCase):
		def setUp(self):
			# THWAP.core.config.thSlurp()
			self.thSlurpFlag = 0
			self.thSlurp = config.thSlurp()

		def thSlurpCallback(self, msg):
			self.thSlurpFlag = 1

		def testslurp(self):
			fp = open('/tmp/thwap', 'w+')
			fp.write('this is a test\nbut this is not\n')
			fp.close()
			fp = open('/tmp/thwap', 'r')
			self.thSlurp.registerTrigger('^this.*', self.thSlurpCallback)
			self.thSlurp.process(fp)
			os.unlink('/tmp/thwap')
			self.assertEqual(self.thSlurpFlag, 1)

		def testconfig(self):
			fp = open('/tmp/thwap', 'w+')
			fp.write('''# Comment
testSection {
	testkey = testvalue
}\n''')
			fp.close()
			self.thConfig = config.thConfig('/tmp/thwap')
			if self.thConfig.conf != {}:
				if self.thConfig.lookup('testSection', 'testkey') == 'testvalue':
					self.thConfig.set('testSection', 'testkey', 'anotherValue')
			self.assertEqual(self.thConfig.lookup('testSection', 'testkey'), 'anotherValue')

if __name__ == '__main__':
	unittest.main()

File: tests/thwap-tests.rb
#!/usr/bin/ruby
require "test/unit"

# Setup our library paths properly so we can be called by make
# or by hand.
if ENV['PWD'] =~ /.*tests.*/
	$:.insert(0, "#{ENV['PWD']}/../ruby/")
else
	$:.insert(0, "#{ENV['PWD']}/ruby/")
end
# imports to test
require "thwap/core/config"
require "thwap/core/threadPools"
def _callback
	return 1
end

# libTHWAP unit tests
class UnitTest < Test::Unit::TestCase
	def test_slurp
		@thSlurpFlag = 0
		fp = File.open('/tmp/thwap', 'w+')
		fp.write("this is a test\nbut this is not")
		fp.close
		fp = File.open('/tmp/thwap', 'r')
		thSlurp = THWAP::Core::ThSlurp.new()
		thSlurp.send :registerTrigger, '^this.*', _callback
		thSlurp.send :process, fp
		assert_equal(0, @thSlurpFlag)
	end

	def test_config
		assert_equal(0, 0)
	end

	def iterator(*args)
		(1..1000000).each do |i|
			a = 1
		end
	end

	def test_threadpools
		obj = THWAP::Core::ThThreadPool.new(10)
		assert_equal(10, obj.threadCount) 
		pool = obj.getThreads
		pool.each do |i|
			i.regWorkLoad(self.iterator, 1)
			i.run
		end
	end
end
